### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by ikhoon**
- Curator 5.6.0 -> 5.7.0
- Dropwizard Metrics 4.2.25 -> 4.2.26
- Eureka 2.0.2 -> 2.0.3
- GraphQL Kotlin 7.1.1 -> 7.1.4
- gRPC Java 1.64.0 -> 1.65.1
- Jackson 2.17.1 -> 2.17.2
- Jetty 9.4.54 -> 9.4.55, 11.0.20 -> 11.0.22, 12.0.8 -> 12.0.12
- Kafka client 3.7.0 -> 3.8.0
- Kubernetes client 6.12.1 -> 6.13.1
- Micrometer 1.13.0 -> 1.13.2
- Micrometer Tracing 1.2.4 -> 1.3.2
- Netty 4.1.110 -> 4.1.112
- Prometheus 1.3.0 -> 1.3.1
- Reactor 3.6.6 -> 3.6.8
- Sangria 4.1.0 -> 4.1.1
- Scala 3.4.1 - 3.4.2
- ScalaPB 0.11.15 -> 0.11.17
- SLF4J 2.0.12 -> 2.0.13
- Spring 6.1.8 -> 6.1.11
- Spring Boot 3.3.0 -> 3.3.2
- Tomcat 9.0.87 -> 9.0.91, 10.1.20 -> 10.1.26
- Build
  - AssertJ 3.25.3 -> 3.26.3
  - Error Prone 2.27.1 -> 2.29.2
  - Error Prone Gradle 3.1.0 -> 4.0.1
  - Fastutil 8.5.13 -> 8.5.14
  - gax-grpc 2.46.1 -> 2.51.0
  - Gradle 8.7 -> 8.9
  - Guava 33.2.0 -> 33.2.1
  - java-websocket 1.5.6 -> 1.5.7
  - JCtools 4.0.3 -> 4.0.5
  - jsoup 1.17.2 -> 1.18.1
  - JUnit 5.10.2 -> 5.10.3
  - NullAway 0.11.0 -> 0.11.1
  - Proguard 7.4.2 -> 7.5.0
  - Testcontainers 1.19.8 -> 1.20.1

### CODE REVIEW

**Comment by ikhoon** [comment]
CI builds were failed due to the same reason:
```
* What went wrong:
Execution failed for task ':core:trimShadedJar'.
> java.io.IOException: Can't read [/home/runner/.gradle/caches/modules-2/files-2.1/org.apache.tomcat.embed/tomcat-embed-core/10.1.26/5e0fa06ca373ef0ca55e603291ea51b590c377ea/tomcat-embed-core-10.1.26.jar] (Can't process class [org/apache/tomcat/util/net/openssl/panama/OpenSSLContext$1.class] (Unsupported version number [66.0] (maximum 65.65535, Java null)))
```

`:core` module does not depend on `tomcat-embed-core` so it seems like a regression. Let me take a closer look first.

**Comment by ikhoon** [comment]
I found out that `OpenSSLContext` is compiled with Java 22 to use The Panama Project. https://apachecon.com/acna2022/slides/03_MAUCHERAT_PAN.pdf

It has been developed in Tomcat 11 but was backported to 10.1.22. https://github.com/apache/tomcat/commit/82dd9eb33fd2f878af393dcfe9209403f66827ec

```
javap -verbose org/apache/tomcat/util/net/openssl/panama/OpenSSLContext |grep major
  major version: 66
```

The problem was fixed by upgrading the Proguard version to 7.5.0 which supports Java 22. https://github.com/Guardsquare/proguard/releases/tag/v7.5

**Comment by ikhoon** [comment]
https://ge.armeria.dev/s/sn5etjaps6hae The leak report will be fixed by https://github.com/line/armeria/pull/5858

File: testing-internal/src/main/java/com/linecorp/armeria/internal/testing/InternalTestingBlockHoundIntegration.java [hunk 1]
```diff
@@ -75,6 +75,7 @@ public void applyTo(Builder builder) {
                 "writeBlockingMethod");
         builder.allowBlockingCallsInside("com.linecorp.armeria.client.ClientFactory", "ofDefault");
         builder.allowBlockingCallsInside("io.envoyproxy.controlplane.cache.SimpleCache", "createWatch");
+        builder.allowBlockingCallsInside("io.grpc.netty.shaded.io.grpc.netty.GrpcHttp2ConnectionHandler", "<clinit>");
```
Comment (ikhoon) [inline testing-internal/src/main/java/com/linecorp/armeria/internal/testing/InternalTestingBlockHoundIntegration.java:78]: `GrpcHttp2ConnectionHandler` changed to read Netty `Version` which is reported as a blocking operation.

https://github.com/grpc/grpc-java/pull/11367/files#diff-ac0ae9f2ac0baa6aad0372713c102af83ea972c6fed7ddcc2767e51dbb372f8fR55

```java
Thread[#42,grpc-default-worker-ELG-3-1,5,main]
java.lang.Exception: java.io.RandomAccessFile#readBytes
        at com.linecorp.armeria.internal.testing.InternalTestingBlockHoundIntegration.writeBlockingMethod(InternalTestingBlockHoundIntegration.java:85)
        at reactor.blockhound.BlockHound$Builder.lambda$install$8(BlockHound.java:488)
        at reactor.blockhound.BlockHoundRuntime.checkBlocking(BlockHoundRuntime.java:89)
        at java.base/java.io.RandomAccessFile.readBytes(RandomAccessFile.java:398)
        at java.base/java.io.RandomAccessFile.read(RandomAccessFile.java:434)
        at java.base/java.util.zip.ZipFile$Source.readAt(ZipFile.java:1541)
        at java.base/java.util.zip.ZipFile$ZipFileInputStream.read(ZipFile.java:956)
        at java.base/java.util.zip.ZipFile$ZipFileInflaterInputStream.fill(ZipFile.java:463)
        at java.base/java.util.zip.InflaterInputStream.read(InflaterInputStream.java:175)
        at java.base/java.io.FilterInputStream.read(FilterInputStream.java:119)
        at java.base/java.io.FilterInputStream.read(FilterInputStream.java:95)
        at java.base/java.util.Properties$LineReader.readLine(Properties.java:504)
        at java.base/java.util.Properties.load0(Properties.java:420)
        at java.base/java.util.Properties.load(Properties.java:409)
        at io.grpc.netty.shaded.io.netty.util.Version.identify(Version.java:77)
        at io.grpc.netty.shaded.io.netty.util.Version.identify(Version.java:56)
        at io.grpc.netty.shaded.io.grpc.netty.GrpcHttp2ConnectionHandler.<clinit>(GrpcHttp2ConnectionHandler.java:55)
        at io.grpc.netty.shaded.io.grpc.netty.NettyServerTransport.createHandler(NettyServerTransport.java:269)
        at io.grpc.netty.shaded.io.grpc.netty.NettyServerTransport.start(NettyServerTransport.java:137)
```

File: dependencies.toml [hunk 2]
```diff
@@ -298,7 +299,7 @@ version.ref = "brotli4j"
 [libraries.bucket4j]
 module = "com.github.vladimir-bukhtoyarov:bucket4j-core"
 version.ref = "bucket4j"
-javadocs = "https://javadoc.io/doc/com.github.vladimir-bukhtoyarov/bucket4j-core/7.6.0/"
+javadocs = "https://javadoc.io/doc/com.github.vladimir-bukhtoyarov/bucket4j-core/7.6.1/"
```
Comment (ikhoon) [inline dependencies.toml:302]:
```suggestion
javadocs = "https://javadoc.io/doc/com.github.vladimir-bukhtoyarov/bucket4j-core/7.6.0/"
```

**Review by jrhee17 — APPROVED** [review]
👍 👍

**Review by minwoox — APPROVED** [review]
👍 👍 👍

**Pull request merged**